Evergreen/MainWindow/MainWindowController.swift

```diff
@@ -69,6 +69,14 @@ class MainWindowController : NSWindowController, NSUserInterfaceValidations {
 		if item.action == #selector(openArticleInBrowser(_:)) {
 			return currentLink != nil
 		}
+        
+        if item.action == #selector(showShareWindow(_:)) {
+            if let link = currentLink {
+                return URL(string: currentLink) != nil
+            } else {
```

Comment:
Right here, you're safely unwrapping the `currentLInk` optional, but then you're never using the safely unwrapped optional.

Also, this check would be a lot simpler written as this.
1. Early termination
2. Safely use unwrapped optional

```swift
    guard let link = currentLink else {
        return false
    }
    return URL(string: link) != nil
```